### PR TITLE
Fix cam dust emiss control

### LINKED ISSUE #2687: Error with setting of prigent streams when coupled with CAM
### Brief summary of bug

F compsets with ctsm5.2.019 (with dust emissions for Leung_2023 in place) fail with an error.

This was originally noted by @fvitt here in CAM:

https://github.com/ESCOMP/CAM/pull/1104#issuecomment-2284303917

### General bug information

**CTSM version you are using:** ctsm5.2.019

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** B and F compsets where CTSM is used coupled to CAM

### Details of bug

This is with cam6_4_015-10-ge393c668 in CAM on @fvitt dust_emissions branch.

### Important details of your setup / configuration so we can reproduce the bug

The problem is that the setting of use_prigent_roughness relies on knowing what dust_emis_method is. When LND_SETS_DUST_EMIS_DRV_FLDS is TRUE this is going to be known by CLM, but not when it's FALSE it needs to get it from CAM and not rely on it's own setting.

### Important output or errors that show the problem

```
cime/scripts> ./create_test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem -r . --walltime 00:20:00
Testnames: ['SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem']
Using project from .cesm_proj: P93300606
create_test will do up to 1 tasks simultaneously
create_test will use up to 160 cores simultaneously
Creating test directory /glade/derecho/scratch/erik/cam_dust_emis/cime/scripts/SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem.20240813_131716_ycbhi8
RUNNING TESTS:
  SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem
Starting CREATE_NEWCASE for test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem with 1 procs
Finished CREATE_NEWCASE for test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem in 9.195398 seconds (PASS)
Starting XML for test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem with 1 procs
Finished XML for test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem in 0.413804 seconds (PASS)
Starting SETUP for test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem with 1 procs
Finished SETUP for test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem in 4.892254 seconds (PASS)
Starting SHAREDLIB_BUILD for test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem with 1 procs
Finished SHAREDLIB_BUILD for test SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem in 6.334215 seconds (FAIL). [COMPLETED 1 of 1]
    Case dir: /glade/derecho/scratch/erik/cam_dust_emis/cime/scripts/SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem.20240813_131716_ycbhi8
    Errors were:
        Building test for SMS in directory /glade/derecho/scratch/erik/cam_dust_emis/cime/scripts/SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem.20240813_131716_ycbhi8
        ERROR: Command /glade/derecho/scratch/erik/cam_dust_emis/components/clm/bld/build-namelist failed rc=255
        out=
        err=Warning : CLM build-namelist::CLMBuildNamelist::setup_logic_prigent_roughness() : use_prigent_roughness does NOT need to on without dust_emis_method being Leung_2023, it simply won't be used
         -- Add -ignore_warnings option to CLM_BLDNML_OPTS to ignore this warning

Waiting for tests to finish
FAIL SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem (phase SHAREDLIB_BUILD)
    Case dir: /glade/derecho/scratch/erik/cam_dust_emis/cime/scripts/SMS_Ln9.f19_f19_mg17.FHIST.derecho_intel.cam-outfrq9s_nochem.20240813_131716_ycbhi8
Due to presence of batch system, create_test will exit before tests are complete.
To force create_test to wait for full completion, use --wait
test-scheduler took 21.338916063308716 seconds
```

### LINKED ISSUE #2709: Problem in megan namelist parser due to an update in the CMEPS parser for megan compounds
### Brief summary of bug

CTSM is failing with a valid megan namelist that CAM is currently using:

``` fortran
&megan_emis_nl
 megan_factors_file		= '/glade/campaign/cesm/cesmdata/inputdata/atm/cam/chem/trop_mozart/emis/megan21_emis_factors_78pft_c20161108.nc'
 megan_mapped_emisfctrs		= .false.
 megan_specifier		= 'SOAE = 0.5954*isoprene + 5.1004*(carene_3 + pinene_a + thujene_a + bornene +', ' terpineol_4 + terpineol_a + terpinyl_ACT_a + myrtenal + sabinene + pinene_b + camphene +',
         ' fenchene_a + limonene + phellandrene_a + terpinene_a + terpinene_g + terpinolene +', ' phellandrene_b + linalool + ionone_b + geranyl_acetone + neryl_acetone + jasmone +',
         ' verbenene + ipsenol + myrcene + ocimene_t_b + ocimene_al + ocimene_c_b + 2met_nonatriene) + ', ' 12.3942*(farnescene_a + caryophyllene_b + acoradiene + aromadendrene + bergamotene_a +',
         ' bergamotene_b + bisabolene_a + bisabolene_b + bourbonene_b + cadinene_d + cadinene_g +', ' cedrene_a + copaene_a + cubebene_a + cubebene_b + elemene_b + farnescene_b +',
         ' germacrene_B + germacrene_D + gurjunene_b + humulene_a + humulene_g + isolongifolene +', ' longifolene + longipinene + muurolene_a + muurolene_g + selinene_b + selinene_d +',
         ' nerolidol_c + nerolidol_t)'
/
```

### General bug information

**CTSM version you are using:** ctsm5.2.019

**Does this bug cause significantly incorrect results in the model's science?** No
**Configurations affected:** Megan namelist with updated format

### Important details of your setup / configuration so we can reproduce the bug

It fails because of the update that came into CMEPS here a year and a half ago...

https://github.com/ESCOMP/CMEPS/commit/9197fd20a6be104c24fd058cae23d2fc1a67670a

This came in with cmeps1.14.12, so came into ctsm in ctsm5.1.120

### Important output or errors that show the problem

```
CLM future scenario SSP-RCP combination is hist
CLM sim_year is 2000
Using clm5_0_cam6.0 for lnd_tuning_mode
ERROR : CLM build-namelist::CLMBuildNamelist::check_megan_spec() : Bad format for megan_specifier = 'SOAE = 0.5954*isoprene + 5.1004*(carene_3 + pinene_a + thujene_a + bornene +'
```

### LINKED ISSUE #2712: Allow user to change drv_flds_in settings even if the relevent (--drydep, --megan, --fire-emiss) option isn't in CLM_BLDNML_OPTS
This is coming up in the context of #2699. And I think I'll likely need to change this behavior at least temporarily. The question is whether this change is a good long term change or not.

There are settings for fields that will go into drv_flds_in (that are needed by both CAM and CTSM) in CLM_BLDNML_OPTS (--drydep, --megan, or fire-emiss) that do these things:

1. Turn on these fields for testing in stand-alone CTSM (with somewhat arbitrary settings determined by CLM).
2. If off (--no-drydep, --no-megan, or --no-fire-emiss) -- don't allow the settings to be set in user_nl_clm.
3. NOTE: If you contradict yourself by setting them both in CAM and CTSM -- it'll die with an error on the contradiction

The first option is important so that we are testing them in stand-alone CTSM in our testing and so that simulations have output for these things that can be examined in the standalone context. The second one isn't strictly required because the 3rd one will ensure that you can't contradict yourself between CAM and CTSM.

My plan is to turn off "2" for now. But, after discussion we can either add it back in or leave it that way depending on what we think is the best way to handle this.

@fvitt

### PR DESCRIPTION
**Pull request by ekluzek**
### Description of changes

Fix dust emissions control when running coupled to CAM.

### Specific notes

Contributors other than yourself, if any:

CTSM Issues Fixed (include github issue #):
 Fixes #2687
 Fixes #2709 
 Fixes #2712

Are answers expected to change (and if so in what way)? No

Any User Interface Changes (namelist or namelist defaults changes)?

Does this create a need to change or add documentation? Did you do so? No

Testing performed, if any:  will be regular
  So far just running namelist testing

### CODE REVIEW

**Comment by ekluzek** [comment]
#2709 blocks the fix for this working.

File: bld/CLMBuildNamelist.pm [hunk 1]
```diff
@@ -1750,6 +1756,12 @@ sub process_namelist_inline_logic {
   ##################################
   setup_logic_lightning_streams($opts,  $nl_flags, $definition, $defaults, $nl);
 
+  ######################################
+  # namelist options for dust emissions
+  ######################################
```
Comment (ekluzek) [inline bld/CLMBuildNamelist.pm:1761]: Add a note that this needs to be done before the other drv_flds_in settings.

Comment (ekluzek) [reply bld/CLMBuildNamelist.pm]: Added some notes around this in 406edd4596fe6bba3d3d5095996fe729d0bccbd3

**Review by slevis-lmwg — APPROVED** [review]
@ekluzek gave me and Will a guided explanation of this PR. This resolves immediate needs with temporary fixes. These will be revisited later with longer term solutions. Thanks, Erik!

**Comment by ekluzek** [comment]
@fvitt and I went over this on a high level and agreed to what's being done here for the short term. I don't like some of what's being done, but think it's OK for the short term. And since this tag needs to happen ASAP we will go with what we have working here.

**Pull request merged**